src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -0,0 +1,522 @@
+    }
+
+    /**
+     * Setter method for token type.
+     * @param tokenType token type
+     */
+    protected void setTokenType(int tokenType) {
+        this.tokenType = tokenType;
+    }
+
+    /**
+     * Setter method for line number.
+     * @param lineNo line number
+     */
+    protected void setLineNo(int lineNo) {
+        this.lineNo = lineNo;
+    }
+
+    /**
+     * Setter method for column number.
+     * @param columnNo column number
+     */
+    protected void setColumnNo(int columnNo) {
+        this.columnNo = columnNo;
+    }
+
+    /**
+     * Getter method for children.
+     * @return children list
+     */
+    protected List<AbstractNode> getChildren() {
+        return Collections.unmodifiableList(children);
+    }
+
+    /**
+     * Add new child node to children list.
+     * @param node child node
+     */
+    protected void addChild(AbstractNode node) {
+        children.add(node);
+    }
+
+    /**
+     * Returns true if nodes are same, false otherwise.
+     * @param nodeInfo other node
+     * @return {@code TreeInfo}
+     */
+    @Override
+    public boolean isSameNodeInfo(NodeInfo nodeInfo) {
+        return this == nodeInfo;
+    }
+
+    /**
+     * Returns if implementation provides fingerprints.
+     * @return {@code boolean}
+     */
+    @Override
+    public boolean hasFingerprint() {
+        return false;
+    }
+
+    /**
+     * Returns uri of the namespace for the current node.
+     * @return uri
+     */
+    // -@cs[AbbreviationAsWordInName] Implements method from the interface.
+    // No way to change method name, because interface is the part of the Saxon library.
+    @Override
+    public String getURI() {
+        return "";
+    }
+
+    /**
+     * Returns if current node has children.
+     * @return if current node has children
+     */
+    @Override
+    public boolean hasChildNodes() {
+        return !children.isEmpty();
+    }
+
+    /**
+     * Determines axis iteration algorithm.
+     * @param axisNumber element from {@code AxisInfo}
+     * @param nodeTest filter for iterator
+     * @return {@code AxisIterator} object
+     */
+    @Override
+    public AxisIterator iterateAxis(byte axisNumber, NodeTest nodeTest) {
+        AxisIterator axisIterator = iterateAxis(axisNumber);
+        if (nodeTest != null) {
+            axisIterator = new Navigator.AxisFilter(axisIterator, nodeTest);
+        }
+        return axisIterator;
+    }
+
+    /**
+     * Determines axis iteration algorithm.
+     * @param axisNumber element from {@code AxisInfo}
+     * @return {@code AxisIterator} object
+     */
+    @Override
+    public AxisIterator iterateAxis(byte axisNumber) {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns line number.
+     * @return line number
+     */
+    // -@cs[SimpleAccessorNameNotation] Implements method from the interface.
+    @Override
+    public int getLineNumber() {
+        return lineNo;
```

Comment:
What is a point of suppressing of Checkstyle ?
It point on the problem, .... why we have field named as lineNo (bad name)? We are in 21 century so all that limitation for small screens and limits on variable names are in past. Please make a naming to let us read code without any comments or javadoc.